src/platforms/hosted/dap_swd.c

```diff
@@ -82,8 +82,9 @@ bool dap_swd_init(adiv5_debug_port_s *target_dp)
 	else
 		target_dp->dp_low_write = NULL;
 	/* Set up the accelerated SWD functions for basic target operations */
+	target_dp->dp_low_read = dap_read_reg_no_check;
 	target_dp->dp_read = dap_dp_read_reg;
-	target_dp->error = dap_swd_clear_error;
+	target_dp->error = firmware_swdp_error;
```

Comment:
This line can be dropped as this function is called via `bmda_swd_dp_init()` which runs just after the DP is already set up in the base ADIv5 implementation.

Reply:
ah, good catch! looks like another method was being redundantly assigned in the jlink also